main.go

```diff
@@ -93,16 +94,24 @@ func main() {
 	flag.BoolVar(&deregisterNEGsOnAnnotationRemoval, "deregister-negs-on-annotation-removal", true, "Deregister NEGs from backend service when annotation removed.")
 	flag.StringVar(&project, "project-id", "", "The project ID of the Google Cloud project where the backend services are created. If not specified, project ID will be fetched from the Metadata server.")
 	flag.BoolVar(&useSvcNeg, "use-svcneg", false, "Use service neg custom resource to get the NEG zone info.")
+
+	// Configure zap logging options with production defaults (JSON format, info level)
 	opts := zap.Options{
-		Development: true,
+		Development: false,
```

Comment:
Maybe this could be bool `--debug`? Not sure how much extra logging the default option will generate...